lib/preprocess.js

```diff
@@ -89,21 +92,33 @@ export const preprocess = async ({
   })
 }
 
-export const fetchMeasurementsViaClient = async (web3Storage, cid) => {
-  const res = await web3Storage.get(cid)
-  const files = await res.files()
-  const measurements = JSON.parse(await files[0].text())
-  return measurements
-}
-
-export const fetchMeasurementsViaGateway = async (cid) => {
-  const res = await fetch(`https://${encodeURIComponent(cid)}.ipfs.w3s.link/measurements.json`)
+export const fetchMeasurementsTrustless = async cid => {
+  const res = await fetch(
+    `https://${encodeURIComponent(cid)}.ipfs.w3s.link?format=car`
+  )
   if (!res.ok) {
     const msg = `Cannot fetch measurements ${cid}: ${res.status}\n${await res.text()}`
     throw new Error(msg)
   }
-  const measurements = await res.json()
-  return measurements
+  const reader = await CarReader.fromIterable(res.body)
+  const entries = exporter(cid, {
+    async get (cid) {
+      const block = await reader.get(cid)
+      await validateBlock(block)
+      return block.bytes
+    }
+  })
+  for await (const entry of entries) {
+    if (entry.type === 'file' || entry.type === 'raw') {
+      const bufs = []
+      for await (const buf of entry.content()) {
+        bufs.push(buf)
+      }
+      const json = Buffer.concat(bufs).toString()
```

Comment:
I think this could be implemented in a more performant way by using [`bl`](https://www.npmjs.com/package/bl), but I guess we are not that much performance/memory-usage sensitive here.

```suggestion
      const bufs = new BufferList()
      for await (const buf of entry.content()) {
        bufs.append(buf)
      }
      const json = bufs.toString()
```